src/master-playlist-controller.js

```diff
@@ -794,23 +794,20 @@ export class MasterPlaylistController extends videojs.EventTarget {
       return this.mediaSource.endOfStream('network');
     }
 
-    // Blacklist this playlist
-    currentPlaylist.excludeUntil = Date.now() + BLACKLIST_DURATION;
+    if (!this.masterPlaylistLoader_.isFinalRendition_()) {
+      // Blacklist this playlist
+      currentPlaylist.excludeUntil = Date.now() + BLACKLIST_DURATION;
 
-    // Select a new playlist
-    nextPlaylist = this.selectPlaylist();
+      // Select a new playlist
+      nextPlaylist = this.selectPlaylist();
 
-    if (nextPlaylist) {
       videojs.log.warn('Problem encountered with the current ' +
                        'HLS playlist. Switching to another playlist.');
 
       return this.masterPlaylistLoader_.media(nextPlaylist);
     }
-    videojs.log.warn('Problem encountered with the current ' +
-                     'HLS playlist. No suitable alternatives found.');
-    // We have no more playlists we can select so we must fail
-    this.error = error;
-    return this.mediaSource.endOfStream('network');
+    // Never blacklisting this playlist because it's final rendition
+    return this.masterPlaylistLoader_.media(currentPlaylist);
```

Comment:
if `currentPlaylist === this.masterPlaylistLoader_.media()` the call to `media()` on the playlist loader will exit early without starting another request for the playlist because it considers it not a change in media. Playlist loader should continue refreshing the live playlist until either it recovers, or the player timesout with a `PLAYER_ERR_TIMEOUT`